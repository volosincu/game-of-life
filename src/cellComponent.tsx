import React, { useContext } from 'react';
import {store} from "./store";

interface CellProps {
    id: number,
    value: boolean
}

function onChangeCell(dispatch: any, id: number, value: boolean) {
    return () => {
        dispatch({type: 'USER_INPUT', payload: {activate: {id: id, value: !value}}})
    };
}

export const CellComponent = React.memo((props: CellProps) => {
    const globalState = useContext(store);
    const { dispatch } = globalState;
    const { id, value } = props;

    return <label className="cell">
        <input type="checkbox" checked={value} onChange={onChangeCell(dispatch, id, value)}></input>
            <span className="cellbox"></span>
    </label>
});
