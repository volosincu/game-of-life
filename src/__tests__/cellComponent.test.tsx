import React from 'react';
import { shallow } from "enzyme";
import { CellComponent } from '../cellComponent';

/**
 @TODO mock store and spy dispatch and onChange event
const mockDispatch = jest.fn();

jest.mock("react", ()=>{
   useContext: () => {
       return {
           dispatch: mockDispatch
       }
   }
});*/


describe("CellComponent component", ()=> {
    let wrapper: any;
    beforeEach(()=> {
        wrapper = shallow(<CellComponent id={1} value={true}/>);
    });

    it('renders correctly - toMatchSnapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the element with \" cell \" className', () => {
        expect(wrapper.find(".cell")).toHaveLength(1);
    });

    it('renders the element with input and span children elements ', () => {
        expect(wrapper.find("input")).toHaveLength(1);
        expect(wrapper.find("span.cellbox")).toHaveLength(1);
    });
});
