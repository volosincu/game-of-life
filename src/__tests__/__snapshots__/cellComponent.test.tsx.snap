// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellComponent component renders correctly - toMatchSnapshot 1`] = `ShallowWrapper {}`;
